src/raftstore/store/peer.rs

```diff
@@ -459,13 +462,42 @@ impl Peer {
             ready.hs.take();
         }
 
+        if let Some(ref soft_state) = ready.ss {
+            if soft_state.leader_id == raft::INVALID_ID {
+                if self.leader_missing_time.is_none() {
+                    self.leader_missing_time = Some(Instant::now())
+                }
+            } else {
+                // When a peer is not initialized, it has no data at storage.
+                // Even if a leader sends heartbeats to it, we consider it as
+                // in the `leader missing` state. That is because if it's isolated from the leader
+                // before it could successfully receive snapshot from the leader and
+                // apply that snapshot, no raft ready event will be triggered,
+                // so that we could not detect the leader is missing for it at here.
```

Comment:
I think it's user's choice to configure `max_leader_missing_duration`. It doesn't have to be at hours or days.

@ngaut @siddontang